sw/snRuntime/src/start.c

```diff
@@ -64,6 +65,7 @@ static inline void snrt_init_bss() {
         size_t size = (size_t)(&__bss_end) - (size_t)(&__bss_start);
         snrt_dma_start_1d_wideptr((uint64_t)(&__bss_start),
                                   (uint64_t)(snrt_zero_memory_ptr()), size);
+        snrt_dma_wait_all();
```

Comment:
There is already a `snrt_dma_wait_all()` call shared for both BSS and CLS transfers in `main()`. Is there a reason for each to have their own?

Performance-wise the previous approach should be better, although it might not make too big of a difference.

For the TLS segment you are right that a wait function call is missing, but we can extend the `main()` function to include `defined(SNRT_INIT_TLS)`. The barrier in `snrt_init_tls()` should also not be needed anymore at that point.

Reply:
Ah I did not see them, in that case they could be removed again. I actually had the issue that one of the DMA transfers took to long and overwrote the section again during the actual execution of the program. But you are right, I will put them in to the `main()` function directly with the others.

Reply:
I tried it out, but a lot of the tests fail when I remove the barrier in the TLS initilization. Therefore, I readded it but removed the redundant `snrt_dma_wait_all` in the BSS and CLS transfers.